### PR TITLE
Java 21 upgrade with Spring Boot 3

### PR DESCRIPTION
**Pull request by zrahman-ch**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -4,4 +4,4 @@ management.endpoints.enabled-by-default=false
 management.endpoints.web.base-path=/
 management.endpoints.web.path-mapping.health=/confirmation-statement-api/healthcheck
```
Comment (dgurjar-ch) [inline src/main/resources/application.properties:5]: Please remove one back slace (/) from base path mapping  .
Like  management.endpoints.web.path-mapping.health=confirmation-statement-api/healthcheck

File: pom.xml [hunk 2]
```diff
@@ -157,6 +161,31 @@
...
+            <plugin>
+                <groupId>com.google.cloud.tools</groupId>
+                <artifactId>jib-maven-plugin</artifactId>
+                <version>${jib-maven-plugin.version}</version>
+                <configuration>
+                    <from>
+                        <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/ci-corretto-build-21:latest</image>
+                    </from>
+                    <to>
+                        <image>169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/confirmation-statement-api:latest</image>
```
Comment (dgurjar-ch) [inline pom.xml:177]: I don't "local" is required, Can you please check. Also I think we should have  will similar image for java and application "416670754337" and same region west-2

File: README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ API service for submitting Confirmation Statements to CHIPS
 
 ### Requirements
 In order to run the service locally you will need the following:
-- [Java 11](https://www.oracle.com/java/technologies/downloads/#java11)
+- [Java 21](https://www.oracle.com/java/technologies/downloads/#java11)
```
Comment (dgurjar-ch) [inline README.md:8]: Please update link as well for java 21, it points to java11

File: pom.xml [hunk 4]
```diff
@@ -5,23 +5,28 @@
...
-        <api-sdk-java.version>4.3.35</api-sdk-java.version>
-        <private-api-sdk-java.version>2.0.313</private-api-sdk-java.version>
+        <java.version>21</java.version>
+        <spring-boot-dependencies.version>3.3.1</spring-boot-dependencies.version>
+        <api-sdk-java.version>6.0.17</api-sdk-java.version>
+        <private-api-sdk-java.version>4.0.76</private-api-sdk-java.version>
         <maven-compiler-plugin.version>3.11.0</maven-compiler-plugin.version>
         <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
         <spring-boot-maven-plugin.version>2.7.18</spring-boot-maven-plugin.version>
         <org.mapstruct.version>1.4.2.Final</org.mapstruct.version>
```
Comment (foman-ch) [inline pom.xml:22]: there's a newer version

File: pom.xml [hunk 5]
```diff
@@ -5,23 +5,28 @@
...
-        <spring-boot-dependencies.version>2.7.11</spring-boot-dependencies.version>
-        <api-sdk-java.version>4.3.35</api-sdk-java.version>
-        <private-api-sdk-java.version>2.0.313</private-api-sdk-java.version>
+        <java.version>21</java.version>
+        <spring-boot-dependencies.version>3.3.1</spring-boot-dependencies.version>
+        <api-sdk-java.version>6.0.17</api-sdk-java.version>
+        <private-api-sdk-java.version>4.0.76</private-api-sdk-java.version>
         <maven-compiler-plugin.version>3.11.0</maven-compiler-plugin.version>
         <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
         <spring-boot-maven-plugin.version>2.7.18</spring-boot-maven-plugin.version>
```
Comment (foman-ch) [inline pom.xml:21]: upgrade to 3.3.1

File: pom.xml [hunk 6]
```diff
@@ -5,23 +5,28 @@
...
-        <java.version>11</java.version>
-        <spring-boot-dependencies.version>2.7.11</spring-boot-dependencies.version>
-        <api-sdk-java.version>4.3.35</api-sdk-java.version>
-        <private-api-sdk-java.version>2.0.313</private-api-sdk-java.version>
+        <java.version>21</java.version>
+        <spring-boot-dependencies.version>3.3.1</spring-boot-dependencies.version>
+        <api-sdk-java.version>6.0.17</api-sdk-java.version>
+        <private-api-sdk-java.version>4.0.76</private-api-sdk-java.version>
         <maven-compiler-plugin.version>3.11.0</maven-compiler-plugin.version>
         <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
```
Comment (foman-ch) [inline pom.xml:20]: upgrade this and the compiler

**Review by foman-ch — CHANGES_REQUESTED** [review]

**Review by foman-ch — APPROVED** [review]

**Comment by foman-ch** [comment]
@aduffy-ch you might want to check why the analyse-pull-request on the pipeline wasn't triggered by this PR

**Pull request merged**